2008-02-21  Corentin Le Gall <clegall@proformatique.com>

	pf_parking_parkdigimute_variable.patch
	$Revision$
	$Date$
	Option for Behavioral change
	
	If set (any value), the channel variable PARKDIGIMUTE disables the
        ast_say_digits() function call, once a call has been parked.
        NB : in the case of an AMI-originated Parking call, the peer who is told
        the '701' digits (for instance), is still not the right one : the
        'parkee' receives this information, while it should be the 'parker'.
        
	Patch status: NOT TESTED on XIVO

Index: asterisk-1.2.27/res/res_features.c
===================================================================
--- asterisk-1.2.27.orig/res/res_features.c	2008-03-19 11:34:11.000000000 +0100
+++ asterisk-1.2.27/res/res_features.c	2008-03-19 11:34:34.000000000 +0100
@@ -282,6 +282,7 @@
 	char exten[AST_MAX_EXTENSION];
         char peername[AST_CHANNEL_NAME];
 	struct ast_context *con;
+        const char *parkdigimute = pbx_builtin_getvar_helper(chan, "PARKDIGIMUTE");
 
 	pu = malloc(sizeof(struct parkeduser));
 	if (!pu) {
@@ -392,7 +393,7 @@
 			ast_log(LOG_ERROR, "Parking context '%s' does not exist and unable to create\n", parking_con);
 		}
 	}
-	if (peer) 
+        if ((peer) && (parkdigimute == NULL))
 		ast_say_digits(peer, pu->parkingnum, "", peer->language);
 	if (con) {
 		snprintf(exten, sizeof(exten), "%d", x);
